### PR TITLE
Benchmarking

### PR DESCRIPTION
**Pull request by movermeyer**
Created the scripts that should allow for easy benchmarking in the future.

Try them out and let me know what you think!

Implements #53

### CODE REVIEW

**Comment by movermeyer** [comment]
Of course, there are even better results for Python 3.7, but I'll add those once I have an official release version [later this month](https://www.python.org/dev/peps/pep-0537/#release-schedule).

**Comment by sdispater** [comment]
I don't think this benchmark is fair since libraries like `arrow` or `pendulum` do much more than what `ciso8601` does.

And if you want to be fair towards `pendulum` you should use its low level parser (with the C extension compiled):

```python
>>> from pendulum.parsing import parse_iso8601
```

See https://pendulum.eustace.io/blog/pendulum-1-2-0-is-out.html#performance-improvements-when-parsing-iso-8601-strings for more information.

And, for this benchmark to be honest it should mention that `ciso8601` is not fully ISO 8601 compliant (see the article above)

Disclaimer: I am the author of pendulum :-)

**Comment by movermeyer** [comment]
Hey @sdispater, I'm happy you saw this. It's cool to hear from you :grinning:

You're right in all regards.

I've been being too strong with my statements. I've added some disclaimers and more references to our [supported subset](https://github.com/closeio/ciso8601#supported-subset-of-iso-8601) of ISO 8601 (Who know what the future holds?: Perhaps we'll get around to extending it to cover the rest someday).

I've also changed the pendulum test to use the lower level parser. `pendulum` sure is fast.  :open_mouth:

File: README.rst [hunk 1]
```diff
@@ -60,77 +64,118 @@ If time zone information is provided, an aware datetime object will be returned.
...
+    +------------+----------+----------+----------+----------+-----------------------------------+
+    |aniso8601   |27.6 usec |28.5 usec |37.7 usec |24.5 usec |110.94x                            |
+    +------------+----------+----------+----------+----------+-----------------------------------+
+    |maya        |54.3 usec |55.4 usec |72.2 usec |56.2 usec |218.47x                            |
+    +------------+----------+----------+----------+----------+-----------------------------------+
+    |arrow       |61.6 usec |65.6 usec |82.8 usec |70.5 usec |247.59x                            |
+    +------------+----------+----------+----------+----------+-----------------------------------+
+    |dateutil    |84.9 usec |88.7 usec |110 usec  |140 usec  |341.35x                            |
+    +------------+----------+----------+----------+----------+-----------------------------------+
+    |moment      |998 msec  |936 msec  |1.09 sec  |1.07 sec  |4.0108e+06x                        |
```
Comment (thomasst) [inline README.rst]: 1 second? Really?

Comment (movermeyer) [reply README.rst]: Yup. :smile: `moment` doesn't do well with timestamps that have timezone info.

Comment (movermeyer) [reply README.rst]: That does seem strange.

Comment (movermeyer) [reply README.rst]: It seems that `moment` only produces a valid datetimes for timezone offsets [that currently exist](https://en.wikipedia.org/wiki/List_of_UTC_time_offsets).

Specifically:

* `xx:00` offsets where xx is in the range [`-12:00`, `+14:00`]
* some `xx:30` offsets (`-04:30`, `-03:30`, `+03:30`, `+04:30`, `+05:30`, `+06:30`, `+09:30`, `+10:30`, `+11:30`)
* some `xx:45` offsets (`+05:45`, +08:45`, +12:45`)

For any other offset, `moment` produces a `None` value, and takes a long time to process.

So `moment` doesn't handle the timezone I supplied, and was taking a long time to return `None`

I'll change the benchmarks to actually check that the correct result is being returned, in the same style as [Pendulem's benchmarking chart](https://pendulum.eustace.io/blog/pendulum-1-2-0-is-out.html#performance-improvements-when-parsing-iso-8601-strings)

**Comment by movermeyer** [comment]
I've added the correctness checks to the benchmarking scripts.

There is still the "issue" that [Pendulum](https://pendulum.eustace.io/) is actually faster for timestamps with timezone information, for Python < 3.7.

I think I can get `ciso8601` faster than it, but that will have to be a different PR.

**Comment by movermeyer** [comment]
Actually, there is another problem: 1.5.1 is not the latest version of `pendulum`.
Though I didn't specify a version, it picked up an old version (Current version is 2.0.2).

This is because `maya` uses an old version of `pendulum`.

I would like to change the way this benchmarking is done to somehow use the latest version of each module.

**Pull request merged**